core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UserCoreUtil.java

```diff
@@ -256,7 +313,7 @@ public static boolean isContain(String name, String[] names) {
      * @return
      * @throws UserStoreException
      */
-    public static String getPolicyFriendlyRandomPassword(String username) throws UserStoreException {
+    public static char[] getPolicyFriendlyRandomPassword(String username) throws UserStoreException {
```

Comment:
This is an API change and therefore can only go with a Major Release (5.0.0) and not minor release (4.5.0).